src/NuGet.Core/NuGet.Protocol.Core.v2/Model/PackageSearchMetadata.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Core.v2
+{
+    public class PackageSearchMetadata : IPackageSearchMetadata
+    {
+        private readonly IPackage _package;
+
+        public PackageSearchMetadata(IPackage package)
+        {
+            if (package == null)
+            {
+                throw new ArgumentNullException(nameof(package));
+            }
+            _package = package;
+        }
+
+        public string Authors => string.Join(", ", _package.Authors);
+
+        public IEnumerable<PackageDependencyGroup> DependencySets
+        {
+            get
+            {
+                return _package.DependencySets.Select(ParsePackageMetadataV2.Convert);
+            }
+        }
+
+        public string Description => _package.Description;
+
+        public long? DownloadCount => _package.DownloadCount;
+
+        public Uri IconUrl => _package.IconUrl;
+
+        public PackageIdentity Identity => new PackageIdentity(_package.Id, V2Utilities.SafeToNuGetVer(_package.Version));
```

Comment:
That's not a regression though. Refactored form here:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs#L98